src/components/NewPerson/NewPerson.tsx

```diff
@@ -0,0 +1,238 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import { Person } from '../../types/person';
+
+import './NewPerson.scss';
+
+type Props = {
+  people: Person[]
+  setPeople: React.Dispatch<React.SetStateAction<Person[]>>
+};
+
+type Errors = {
+  name: boolean
+  sex: boolean
+  born: boolean
+  died: boolean
+};
+
+const currYear = new Date().getFullYear();
+
+const NewPerson: React.FC<Props> = ({ people, setPeople }) => {
+  const [name, setName] = useState('');
+  const [sex, setSex] = useState('');
+  const [born, setBorn] = useState(1400);
+  const [died, setDied] = useState(1400);
+  const [motherName, setMotherName] = useState('');
+  const [fatherName, setFatherName] = useState('');
+
+  const [touched, setTouched] = useState({
+    name: false,
+    sex: false,
+    born: false,
+    died: false,
+  });
+
+  const clearInputs = () => {
+    setName('');
+    setSex('');
+    setBorn(1400);
+    setDied(1400);
+    setMotherName('');
+    setFatherName('');
+  };
+
+  const clearErrors = () => {
+    setTouched({
+      name: false,
+      sex: false,
+      born: false,
+      died: false,
+    });
+  };
+
+  const onFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const newPerson = {
+      name,
+      sex,
+      born,
+      died,
+      motherName,
+      fatherName,
+      slug: `${name.toLowerCase().replace(' ', '-')}-${born}`,
+      father: people.find(person => person.name === fatherName),
+      mother: people.find(person => person.name === motherName),
+    };
+
+    clearInputs();
+    clearErrors();
+
+    setPeople([...people, newPerson]);
+  };
+
+  const validate = () => ({
+    name: name.length === 0 || !/^[A-Za-z\s]*$/.test(name),
+    sex: sex.length === 0,
+    born: born < 1400 || born > currYear,
+    died: died < born || died >= born + 150,
+  });
+
+  const mothers = people.filter(person => person.sex === 'f' && person.born < born + 16);
+  const fathers = people.filter(person => person.sex === 'm' && person.born < born + 16);
```

Comment:
why +16?

Reply:
i supposed that mothers and fathers of kids should be at least 16 years older)